### PR TITLE
[PDI-18784] Ensure the same transformation doesn't get cached multipl…

### PR DESCRIPTION
**Pull request by peterrinehart**
…e times.

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 3m 13s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,core
```
---
:ok_hand: All tests passed!
 
Tests run: **1167**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/metaverse-plugin/job/PR-624/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,core
```
---
:ok_hand: All tests passed!
 
Tests run: **1167**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/metaverse-plugin/job/PR-624/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 14s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,core
```
---
:ok_hand: All tests passed!
 
Tests run: **1167**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/metaverse-plugin/job/PR-624/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: api/src/main/java/org/pentaho/metaverse/api/analyzer/kettle/ExternalResourceCache.java [hunk 1]
```diff
@@ -162,6 +163,18 @@ public String toString() {
   }
 
   public class TransValues extends Resources<Trans> {
+
+    @Override
+    public void add( final Trans value ) {
+      List<Trans> internalList = new ArrayList<>();
+      internalList.addAll( internal );
+      for ( Trans t : internalList ) {
```
Comment (NJtwentyone) [inline api/src/main/java/org/pentaho/metaverse/api/analyzer/kettle/ExternalResourceCache.java:171]: Why are you creating a new list? You could just iterate with the original internal list like in line 140
"for ( IExternalResourceInfo resource : internal ) {"

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 9s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl api,core
```
---
:ok_hand: All tests passed!
 
Tests run: **1167**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/metaverse-plugin/job/PR-624/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request closed without merging**